packages/client/hmi-client/src/utils/number.ts

```diff
@@ -68,3 +71,58 @@ export function displayNumber(num: string): string {
 	if (digitString.length > 6) return parseFloat(num).toExponential(3);
 	return numberToNist(num);
 }
+
+export function maskToNistNumber(input: string): string {
+	// Remove any characters that are not digits, '.', '+', '-', or 'e'
```

Comment:
This function feels like it is halfway between parsing, formatting, and validating ... which means maybe we are over-thinking this business. People don't usually go about writing their own number parsers, cause the functions are usually built-in .... so we should have a sync up on what we are really after here.

I am likely missing some hidden utilities here, but it sort of feels like I can condense the function down to just this:
```
export function maskToNistNumber(input: string): string {
  let maskedInput = input.replace(/\s/g, '');
  return numberToNist(maskedInput);
}
```

For the parsing/validation part, it will accept `4e3e2e1` which seems quite odd.

Reply:
I did try this - how there are many cases where this will not work since this is for input.  i.e if the user wanted decimal places they would type in `123.` and parseFloat would parse out the decimal and the input would be left at just `123` without the user ever being about to type in a decimal place.